### PR TITLE
chore: fix all input processors

### PR DESCRIPTION
**Pull request by jonaslagoni**
**Description**
This PR fixes all the processor tests.

### CODE REVIEW

File: src/processors/JsonSchemaInputProcessor.ts [hunk 1]
```diff
@@ -14,16 +14,35 @@ export class JsonSchemaInputProcessor extends AbstractInputProcessor {
...
+        throw "Input is not a JSON Schema, so it cannot be processed."
+    }
+
+
+    /**
+     * Unless the schema states one that is not supported we assume its of type JSON Schema
+     * 
+     * @param input 
+     */
+    isForMe(input: any): boolean {
```
Comment (jonaslagoni) [inline src/processors/JsonSchemaInputProcessor.ts:36]: I am not sure I like the name... But cant come up with something else.

File: src/processors/InputProcessor.ts [hunk 2]
```diff
@@ -1,21 +1,19 @@
...
+export class InputProcessor {
+  processors: {[key: string]: AbstractInputProcessor} = {};
 
   constructor(){
-    super();
     const asyncAPI = new AsyncAPIInputProcessor();
     this.addProcessor("asyncapi", asyncAPI); 
     const jsonSchema = new JsonSchemaInputProcessor();
-    this.addProcessor("json-schema", jsonSchema);
+    this.addProcessor("default", jsonSchema);
```
Comment (jonaslagoni) [inline src/processors/InputProcessor.ts:16]: I could not find any other way to add a default processor while still making it changeable.

**Review by jonaslagoni — COMMENTED** [review]
Do you think it makes sense to have the `InputProcessor` which simplifies everything where you dont need to know the input type?

**Review by magicmatatjahu — CHANGES_REQUESTED** [review]

File: src/processors/AsyncAPIInputProcessor.ts [hunk 3]
```diff
@@ -31,7 +32,7 @@ export class AsyncAPIInputProcessor extends AbstractInputProcessor {
      * 
      * @param input 
      */
-    static isAsyncAPI(input: any) : boolean {
+    isForMe(input: any) : boolean {
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/processors/AsyncAPIInputProcessor.ts:35]: Agreed!

File: src/processors/InputProcessor.ts [hunk 4]
```diff
@@ -1,21 +1,19 @@
...
-
 import { CommonInputModel } from "../models/CommonInputModel";
 
 /**
  * Main input processor which figures out the type of input it receives and delegates the processing into separate individual processors.
  */
-export class InputProcessor extends AbstractInputProcessor {
-  processors: Map<string, AbstractInputProcessor> = new Map();
+export class InputProcessor {
+  processors: {[key: string]: AbstractInputProcessor} = {};
```
[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/processors/InputProcessor.ts:10]: I hate working with Map... 🤷

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply src/processors/InputProcessor.ts]: Well then

[the remark this answers is not in the document]
Comment (jonaslagoni) [reply src/processors/InputProcessor.ts]: Nwm, I just thought I could not iterate entries easily, that was a lie, changing it back 😄

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply src/processors/InputProcessor.ts]: You can make `for (const [key, value] of processors) {}`

File: src/generators/AbstractGenerator.ts [hunk 5]
```diff
@@ -20,24 +20,25 @@ export const defaultGeneratorOptions = {
  * Abstract generator which must be implemented by each language
  */
 export abstract class AbstractGenerator<O = object> {
+  processor = new InputProcessor();
```
[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply src/generators/AbstractGenerator.ts:23]: You didn't change it :P

**Review by magicmatatjahu — APPROVED** [review]

**Pull request merged**